scripts/create_static_narrative.py

```diff
@@ -0,0 +1,203 @@
+
+import argparse
+import logging
+import sys
+from configparser import ConfigParser
+from os import environ
+from typing import Any
+
+from lib.installed_clients.WorkspaceClient import Workspace
+from lib.StaticNarrative.creator import StaticNarrativeCreator
+from lib.StaticNarrative.narrative_ref import NarrativeRef
+
+DEPLOY = "KB_DEPLOYMENT_CONFIG"
+SERVICE = "KB_SERVICE_NAME"
+AUTH = "auth-service-url"
+
+
+def get_config_file() -> str:
+    """Get the path to the config file.
+
+    :return: path to the config file
+    :rtype: str
+    """
+    return environ.get(DEPLOY, "./test/deploy.cfg")
+
+
+def get_service_name() -> str:
+    """Get the service name.
+
+    :return: service name
+    :rtype: str
+    """
+    return environ.get(SERVICE, "StaticNarrative")
+
+
+def get_config() -> None | dict[str, str]:
+    """Retrieve the config from the config file.
+
+    :return: parsed config as a dictionary
+    :rtype: None | dict[str, str]
+    """
+    if not get_config_file():
+        return None
+    retconfig = {}
+    config = ConfigParser()
+    config.read(get_config_file())
+    for nameval in config.items(get_service_name() or "StaticNarrative"):
+        retconfig[nameval[0]] = nameval[1]
+    return retconfig
+
+
+config = get_config()
+
+
+class StaticNarrativeCmdLine:
+    """Class for creating static narratives from the command line."""
+
+    def __init__(self: "StaticNarrativeCmdLine", config: dict[str, str]) -> None:
+        """Init the class.
+
+        :param self: this class
+        :type self: StaticNarrativeCmdLine
+        :param config: parsed config
+        :type config: dict[str, str]
+        """
+        logging.basicConfig(
+            format="%(created)s %(levelname)s: %(message)s", level=logging.INFO
+        )
+        self.logger = logging.getLogger("StaticNarrative")
+        self.logger.setLevel(logging.INFO)
+        ch = logging.StreamHandler()
+        ch.setLevel(logging.INFO)
+        formatter = logging.Formatter("%(asctime)s - %(levelname)s - %(message)s")
+        ch.setFormatter(formatter)
+        self.logger.addHandler(ch)
+        self.config = config
+
+    def get_narrative_id_from_workspace(
+        self: "StaticNarrativeCmdLine", ws_ref: str | int, token: str
+    ) -> str:
+        """Retrieve the Narrative object from a workspace (if one exists).
+
+        :param self: this class
+        :type self: StaticNarrativeCmdLine
+        :param ws_ref: workspace reference (ID or name)
+        :type ws_ref: str | int
+        :param token: user token
+        :type token: str
+        :raises ValueError: if no Narrative object is found
+        :return: KBaseNarrative.Narrative object UPA
+        :rtype: str
+        """
+        ws_args = "workspaces"
+        if str(ws_ref).isdigit():
+            ws_args = "ids"
+
+        ws_client = Workspace(self.config["workspace-url"], token=token)
+        results = ws_client.list_objects(
```

Comment:
Another way to get the narrative object id is from the workspace info - it should have a metadata key "narrative" which is the id of the narrative that gets worked on.

That might be helpful if there's the odd case of a workspace with multiple narratives. Probably, YAGNI, so this is fine.

Reply:
Ah -- that sounds like a better way of doing things. I think that there are something like 70-odd workspaces with more than one narrative object; I'll ask Jason what was happening with them but I think they were just going to be left as-is.

Reply:
I'm punting on this for now -- I want to first check what's going on with the multi-narrative object workspaces before I implement it... JUST IN CASE!